haystack/nodes/reader/table.py

```diff
@@ -116,68 +137,164 @@ def predict(self, query: str, documents: List[Document], top_k: Optional[int] =
                                     return_tensors="pt",
                                     truncation=True)
             inputs.to(self.devices[0])
-            # Forward query and table through model and convert logits to predictions
-            outputs = self.model(**inputs)
-            inputs.to("cpu")
-            if self.model.config.num_aggregation_labels > 0:
-                aggregation_logits = outputs.logits_aggregation.cpu().detach()
-            else:
-                aggregation_logits = None
 
-            predicted_output = self.tokenizer.convert_logits_to_predictions(
-                inputs,
-                outputs.logits.cpu().detach(),
-                aggregation_logits
-            )
-            if len(predicted_output) == 1:
-                predicted_answer_coordinates = predicted_output[0]
-            else:
-                predicted_answer_coordinates, predicted_aggregation_indices = predicted_output
+            if isinstance(self.model, TapasForQuestionAnswering):
+                current_answer = self._predict_tapas_for_qa(inputs, document)
+                answers.append(current_answer)
+            elif isinstance(self.model, self.TapasForScoredQA):
+                current_answers = self._predict_tapas_for_scored_qa(inputs, document)
+                answers.extend(current_answers)
 
-            # Get cell values
-            current_answer_coordinates = predicted_answer_coordinates[0]
-            current_answer_cells = []
-            for coordinate in current_answer_coordinates:
-                current_answer_cells.append(table.iat[coordinate])
+        # Sort answers by score and select top-k answers
+        if isinstance(self.model, self.TapasForScoredQA):
+            # Answers are sorted first by the general score for the tables and then by the answer span score
+            answers = sorted(answers, reverse=True, key=lambda ans: (ans.meta["table_score"], ans.score))  # type: ignore
```

Comment:
So in the sorted list of Answers that the reader returns the first `Answer.score` might be lower than the second one? If yes, that might be confusing to show to end users ("Why has the first result a lower score?"). Maybe we can combine table_score and answer_score somehow?